packages/vscode-extension/src/project/applicationSession.ts

```diff
@@ -57,19 +57,6 @@ interface LaunchApplicationSessionDeps {
   devtoolsPort?: number;
 }
 
-function waitForAppReady(inspectorBridge: RadonInspectorBridge, cancelToken?: CancelToken) {
-  // set up `appReady` promise
-  const { promise, resolve, reject } = Promise.withResolvers<void>();
-  cancelToken?.onCancel(() => {
-    reject(new CancelError("Cancelled while waiting for the app to be ready."));
-  });
-  const appReadyListener = inspectorBridge.onEvent("appReady", resolve);
-  promise.finally(() => {
-    appReadyListener.dispose();
-  });
-  return promise;
-}
```

Comment:
You should be able to keep this helper function by doing
```
finally {
  appReadyPromise.catch(noop)
  /* ... */
}
```
instead of 
```
finally {
  resolveAppReady();
  /* ... */
}
```
And actually, if you just want to suppress the "Uncaught promise rejection", I think you could just do `promise.catch(noop)` in this function and return `promise` (without `.catch`) -- then there's some rejection handler, but we still get the error outside of this function.